### ISSUE TITLE
checksec in the debugger

### ISSUE THREAD

**Issue by radare** [radare/radare2#2218]
r2 needs to provide checksec information from the libraries loaded in memory. we already can extract this information from a specific module. and this is very useful when doing exploiting. So we must display that information in `dm`. or maybe just as a separate commnad implemented in a plugin.

FMI: https://github.com/philwantsfish/gdb_commands

**Comment by jvoisin** [comment]
We already have some informations in `i`, like ASLR, RELRO, canary, …

**Comment by radare** [comment]
Yes. But you have to go lib by lib in hand to automatize this and extract the info from all the libs loaded in memory. So this can be done with a script using #!pipe (choose python or js) its an easy task, probably suitable for rsoc

**Comment by Maijin** [comment]
R2-pipe python must implement an open in debugging mode before doing this

**Comment by radare** [comment]
Wat?

**Comment by radare** [comment]
What makes you think this is not already done?

**Comment by Maijin** [comment]
? https://github.com/radare/radare2-bindings/tree/master/r2pipe/python

**Comment by radare** [comment]
#!pipe python foo.py

In foo.py just do:

R= r2pipe.open('#!pipe')
DbgMaps = R.cmdj("dmj")

**Comment by radare** [comment]
well you can't spawn a debugger session with r2pipe directly now, do you
want to propose a new .debug() method instead of the .open() one? bear
in mind that we can also statically debug with ESIL.

**Comment by jvoisin** [comment]
I don't think that I would call this task *easy*.

**Comment by XVilka** [comment]
@jvoisin this was you who added this label.

**Comment by dukebarman** [comment]
Will be good if we can use option in rabin2 for print only security info like in checksec:
```
$ checksec.sh --file /bin/ls
RELRO           STACK CANARY      NX            PIE             RPATH      RUNPATH      FILE
Partial RELRO   Canary found      NX enabled    No PIE          No RPATH   No RUNPATH   /bin/ls
```
And if we use own names for this options than add old (from checksec) names near in `()` symbols

**Comment by radare** [comment]
Which one do u propose? Isnt that already in -I?

**Comment by dukebarman** [comment]
output for RELRO like in checksec, for example, and may be change sort order - this options will be place first

**Comment by dukebarman** [comment]
@monosource in his blog post used filter with `~`

```
[0x7f5082528cc0]> i~pic,canary,nx,crypto,stripped,static,relocs
pic      false
canary   true
nx       true
crypto   false
stripped true
static   false
relocs   false
```

It's a same thing that I propose above, so maybe add option (new symbol) for `i` command

**Comment by Maijin** [comment]
Need a letter ?

**Comment by dukebarman** [comment]
Need a command like checksec :)

**Comment by radare** [comment]
Why?

**Comment by alvarofe** [comment]
Don't forget `ik~relro` that is saved in sdb. The info is already there i don't see the need of a new command for that either create a macro or r2pipe to get info with the desired format.

**Comment by radare** [comment]
This is also related to rbin support for loading binary headers from memory

**Comment by ret2libc** [comment]
This issue has been moved from radareorg/radare2 to radareorg/ideas as we are trying to clean our backlog and this issue has probably been created a long while ago. This is an effort to help contributors understand what are the actionable items they can work on, prioritize issues better and help users find active/duplicated issues more easily. If this is not an enhancement/improvement/general idea but a bug, feel free to ask for re-transfer to main repo. Thanks for your understanding and contribution with this issue.
